### PR TITLE
[20709] Make rtps messages headers private

### PR DESCRIPTION
**Pull request by elianalf**
## Description

Make rtps messages headers private :
* RTPSMessageGroup.h
* RTPSMessageCreator.h
* CDRMessage.h 
* CDRMessage.hpp -> CDRMessage.cpp

Related PR:
*  Fast DDS Python: eProsima/Fast-DDS-python/pull/117
* Fast DDS Docs: eProsima/Fast-DDS-docs/pull/744
* Shapes Demo: eProsima/ShapesDemo/pull/136

## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- N/A Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- N/A Any new/modified methods have been properly documented using Doxygen. 
- ❌ Changes are ABI compatible. 
- ❌ Changes are API compatible. 
- N/A New feature has been added to the `versions.md` file (if applicable).
- N/A New feature has been documented/Current behavior is correctly described in the documentation. 
    
- N/A Applicable backports have been included in the description.

## Reviewer Checklist
- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- [x] Check CI results: changes do not issue any warning.
- [x] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

File: include/fastdds/dds/core/policy/ParameterTypes.hpp [hunk 1]
```diff
@@ -28,7 +28,7 @@
 
 #include <fastdds/rtps/common/all_common.h>
 #include <fastdds/rtps/common/Token.h>
-#include <fastdds/rtps/messages/CDRMessage.h>
+#include <fastdds/rtps/common/CDRMessage_t.h>
```
Comment (EduPonz) [inline include/fastdds/dds/core/policy/ParameterTypes.hpp:31]: The struct is forward declared, so we can remove this include
```suggestion
```

File: include/fastdds/rtps/messages/RTPSMessageSenderInterface.hpp [hunk 2]
```diff
@@ -26,7 +26,7 @@
 #include <vector>
 
 #include <fastdds/rtps/common/Guid.h>
-#include <fastdds/rtps/messages/CDRMessage.h>
+#include <fastdds/rtps/common/CDRMessage_t.h>
```
Comment (EduPonz) [inline include/fastdds/rtps/messages/RTPSMessageSenderInterface.hpp:29]: I'd rather have a fwd declaration
```suggestion
```

**Comment by elianalf** [comment]
@richiprosima test linux

File: include/fastdds/rtps/transport/test_UDPv4TransportDescriptor.h [hunk 3]
```diff
@@ -16,10 +16,11 @@
 #define _FASTDDS_TEST_UDPV4_TRANSPORT_DESCRIPTOR_
 
 #include <functional>
+#include <atomic>
 
 #include <fastdds/rtps/transport/SocketTransportDescriptor.h>
 #include <fastdds/rtps/common/SequenceNumber.h>
-#include <fastdds/rtps/messages/CDRMessage.h>
+#include <fastdds/rtps/common/CDRMessage_t.h>
```
Comment (EduPonz) [inline include/fastdds/rtps/transport/test_UDPv4TransportDescriptor.h:23]: Fwd declaration
```suggestion
```

File: src/cpp/fastdds/core/policy/ParameterList.hpp [hunk 4]
```diff
@@ -23,8 +23,8 @@
 #include <fastdds/dds/core/policy/ParameterTypes.hpp>
 #include <fastdds/rtps/messages/RTPS_messages.h>
 #include <fastdds/rtps/common/CDRMessage_t.h>
-#include <fastdds/rtps/messages/CDRMessage.h>
 #include <fastdds/rtps/common/CacheChange.h>
+#include <rtps/messages/CDRMessage.h>
```
Comment (EduPonz) [inline src/cpp/fastdds/core/policy/ParameterList.hpp:27]: Please apply everywhere. Space between public and private includes
```suggestion

#include <rtps/messages/CDRMessage.h>
```

File: src/cpp/rtps/messages/CDRMessage.h
Comment (EduPonz) [inline src/cpp/rtps/messages/CDRMessage.h:1]: Rename to .hpp

File: src/cpp/rtps/messages/CDRMessage.h [hunk 5]
```diff
@@ -350,7 +350,7 @@ inline bool skip(
 
 #ifndef DOXYGEN_SHOULD_SKIP_THIS
 
-#include <fastdds/rtps/messages/CDRMessage.hpp>
+#include "CDRMessage.cpp"
```
Comment (EduPonz) [inline src/cpp/rtps/messages/CDRMessage.h:353]:
```suggestion
```

File: src/cpp/rtps/messages/RTPSMessageCreator.h
Comment (EduPonz) [inline src/cpp/rtps/messages/RTPSMessageCreator.h:1]: Rename to .hpp

File: src/cpp/rtps/messages/RTPSMessageGroup.h
Comment (EduPonz) [inline src/cpp/rtps/messages/RTPSMessageGroup.h:1]: Raname to .hpp

File: src/cpp/rtps/messages/RTPSMessageGroup.h [hunk 6]
```diff
@@ -22,7 +22,7 @@
 #ifndef DOXYGEN_SHOULD_SKIP_THIS_PUBLIC
 
 #include <fastdds/rtps/messages/RTPSMessageSenderInterface.hpp>
-#include <fastdds/rtps/messages/RTPSMessageCreator.h>
+#include <rtps/messages/RTPSMessageCreator.h>
```
Comment (EduPonz) [inline src/cpp/rtps/messages/RTPSMessageGroup.h:25]: Fix include order

File: src/cpp/security/authentication/PKIDH.cpp [hunk 7]
```diff
@@ -21,16 +21,15 @@
...
+
+#include <fastdds/core/policy/ParameterList.hpp>
 #include <fastdds/rtps/security/logging/Logging.h>
 #include <fastdds/dds/log/Log.hpp>
-#include <fastdds/rtps/messages/CDRMessage.h>
 #include <fastdds/rtps/builtin/data/ParticipantProxyData.h>
-
+#include <rtps/messages/CDRMessage.h>
+#include <security/authentication/PKIIdentityHandle.h>
 #include <openssl/opensslv.h>
```
Comment (EduPonz) [inline src/cpp/security/authentication/PKIDH.cpp:31]: Fix include order with external headers

File: test/mock/rtps/RTPSMessageGroup/rtps/messages/RTPSMessageGroup.h
Comment (EduPonz) [inline test/mock/rtps/RTPSMessageGroup/rtps/messages/RTPSMessageGroup.h:1]: You'd need to rename this too

**Review by EduPonz — CHANGES_REQUESTED** [review]

**Comment by elianalf** [comment]
@richiprosima test linux

File: src/cpp/rtps/messages/CDRMessage.h (hunk 5 above)
Comment (elianalf) [reply src/cpp/rtps/messages/CDRMessage.h:353]: It fails to build without it and I think it's because in the .cpp there are all inline functions

**Comment by elianalf** [comment]
@richiprosima test linux

File: src/cpp/rtps/messages/CDRMessage.h (hunk 5 above)
Comment (EduPonz) [reply src/cpp/rtps/messages/CDRMessage.h]: I think it's because the cpp is not included as a source file for the targets in the CMakeLists.txt files, you'd need to do that. I'd also remove the `inline` specifier from the functions in the .cpp

**Comment by elianalf** [comment]
@richiprosima test windows

**Comment by elianalf** [comment]
@richiprosima please test_3 this

File: src/cpp/fastdds/core/policy/ParameterList.hpp
Comment (EduPonz) [inline src/cpp/fastdds/core/policy/ParameterList.hpp:1]: Fix include order

File: src/cpp/rtps/messages/CDRMessage.cpp [hunk 8]
```diff
@@ -19,7 +19,7 @@
  *
  */
 
-#include "CDRMessage.hpp"
+#include <rtps/messages/CDRMessage.hpp>
```
Comment (EduPonz) [inline src/cpp/rtps/messages/CDRMessage.cpp:22]: I'd revert this change. Pre-processor is faster the other way, and they are side by side

File: src/cpp/security/authentication/PKIDH.cpp (hunk 7 above)
Comment (EduPonz) [reply src/cpp/security/authentication/PKIDH.cpp]: So the order is:

1. Corresponding header (for .cpp files)
2. C headers
3. C++ standard headers
4. External headers
5. Same project public headers
6. Internal headers

**Review by EduPonz — CHANGES_REQUESTED** [review]
Minor NITs. There are some still open points as well

**Comment by elianalf** [comment]
@richiprosima please test_3 this

**Review by EduPonz — APPROVED** [review]
